frontend/components/ModelForm.tsx

```diff
@@ -46,6 +46,19 @@ type TableField = {
   label?: string
   name: string
   converter?: (field: any, object: ModelObject) => any
+  render?: (field: any, object: ModelObject) => any
+  id?: string
+}
+
+type Option = {
+  label: string
+  key: any
+}
+
+type FilterOption = {
+  options: Option[]
+  label: string
+  filterFunction: (a, b) => boolean
```

Comment:
Might be a good idea to add type annotation/more concrete argument description here, but this is fine too!